plugins/main/ipvlan/ipvlan.go

```diff
@@ -145,13 +145,14 @@ func createIpvlan(conf *NetConf, ifName string, netns ns.NetNS) (*current.Interf
 		Mode: mode,
 	}
 
-	if err := netlink.LinkAdd(mv); err != nil {
+	if err := netlink.LinkAdd(iv); err != nil {
 		return nil, fmt.Errorf("failed to create ipvlan: %v", err)
 	}
 
 	err = netns.Do(func(_ ns.NetNS) error {
 		err := ip.RenameLink(tmpName, ifName)
 		if err != nil {
+			_ = netlink.LinkDel(iv)
```

Comment:
IMO, if we want to avoid link leaking, all possible errors should be walked, e.g. [https://github.com/containernetworking/plugins/blob/48a97a7ad16cb839d4871cd493faca13129737ca/plugins/main/ipvlan/ipvlan.go#L162](https://github.com/containernetworking/plugins/blob/48a97a7ad16cb839d4871cd493faca13129737ca/plugins/main/ipvlan/ipvlan.go#L162)

Reply:
I suspect that this kind of cleanup error is difficult and not that important to handle.  Give it a logging or aggregate multiple errors then return? Any thoughts?

Reply:
Could you put link recycling in `defer`? And delete link by name?